pkgs/development/compilers/gcc/8/default.nix

```diff
@@ -119,14 +119,15 @@ let version = "8.3.0";
             "--disable-decimal-float" # No final libdecnumber (it may work only in 386)
         ]));
     stageNameAddon = if crossStageStatic then "-stage-static" else "-stage-final";
-    crossNameAddon = if targetPlatform != hostPlatform then "-${targetPlatform.config}" + stageNameAddon else "";
+    crossNameAddon = if targetPlatform != hostPlatform then "${targetPlatform.config}" + stageNameAddon + "-" else "";
 
     bootstrap = targetPlatform == hostPlatform;
 
 in
 
 stdenv.mkDerivation ({
-  name = "${name}${if stripped then "" else "-debug"}-${version}" + crossNameAddon;
+  pname = crossNameAddon + "${name}${if stripped then "" else "-debug"}";
```

Comment:
I see we were missing this as 2afd3c901e4d1038ae0ba93bd3765002b997e815 and 659363fb401a9916f8aa92f297f96c0584432dc7 must have crossed.

Reply:
I messed this up recently too. Maybe we can finally fix it in this PR.